rfcs/20220105-sig-go.md

```diff
@@ -0,0 +1,81 @@
+[github.com/tensorflow/tensorflow/go](https://github.com/tensorflow/tensorflow/tree/r2.8/tensorflow/go).
+These bindings are
+[loosely integrated with the build system](https://github.com/tensorflow/tensorflow/pull/50934)
+and are presently listed as being both
+[unsupported by the community](https://www.tensorflow.org/versions) and
+[unsupported by the TensorFlow team](https://github.com/tensorflow/build/tree/master/golang_install_guide).
+
+Significant problems presently exist in the usability of TensorFlow's Go
+bindings.  Changes to how Go manages dependencies, specifically in the
+implementation of [Go Modules](https://blog.golang.org/using-go-modules), have
+resulted in a set of installation challenges (see, for instance, GitHub issues
+[41808](https://github.com/tensorflow/tensorflow/issues/41808) and
+[43847](https://github.com/tensorflow/tensorflow/issues/43847)). To alleviate
+these issues, there has been discussion regarding relocation of the Go bindings
+to a dedicated repository
+([1](https://github.com/tensorflow/tensorflow/pull/44655#issuecomment-725040183),
+[2](https://github.com/tensorflow/tensorflow/pull/50934)).  A component of this
+SIG proposal is a request for TensorFlow team support and collaboration in
+migrating the TensorFlow Go bindings to a dedicated GitHub repository.
+
+This request proposes that an opportunity exists to establish a effective SIG
+around the Go bindings. In the past, the TensorFlow developer community has
+made valuable contributions to the Go bindings, but community contributions
+have been sporadic. Implicit in this request is an acknowledgment that the
+current approach is unlikely to fully promote community collaboration due, for
+instance, to collocation of the bindings with TensorFlow core, methods for
+issue assignment/handling, the current state of the bindings in terms of
+usability, and the recent shift of the bindings into unsupported status.  The
+expected outcome of establishing SIG Go is to increase community collaboration
+around the Go bindings, leading to the resolution of these issues and a pathway
+for long-term maintenance and development activities.
+
+## Goals & Objectives
+
+The SIG proposes a charter focused on conservative aims: engaging the
+community, solving current and future usability issues, performing on-going
+maintenance, and making incremental improvements to the existing codebase
+(e.g., functionality, testing, documentation).
+
+More specifically, SIG Go's initial focus will be on the following set of
+problems/objectives:
+
+* Fostering community involvement for the project and membership in SIG Go.
+* Migrating TensorFlow/Go source code from the main TensorFlow repository to
+  a dedicated repository.
+* Restoring usability of the bindings, specifically achieving interoperability
+  with Go tooling for module installation.
+* Improving unit testing and the coverage of tests.
+* Improving package documentation, including example code.
```

Comment:
There is also https://github.com/tensorflow/examples not sure if we should link from there or what is the best way to make things discoverable.

Reply:
I was thinking along the lines of [testable examples](https://go.dev/blog/examples) here -- examples demonstrating use of the API.  These are conveniently published alongside the package documentation.

Reply:
good points, these as tests, should definitely live with code. What about bigger examples more like tutorials or end-to-end examples?

Reply:
These should become a distinct/additional item on the list here (rfc) and corresponding Launch Plan within the SIG Request, imho.  Good add.  I'm not sure what the best location is.